src/conversions.jl

```diff
@@ -677,22 +677,29 @@ convert_attribute(A::AbstractVector, ::key"linestyle") = A
 """
     A `Symbol` equal to `:dash`, `:dot`, `:dashdot`, `:dashdotdot`
 """
-function convert_attribute(ls::Symbol, ::key"linestyle")
-    return if ls == :solid
+convert_attribute(ls::Union{Symbol,AbstractString}, ::key"linestyle") = pattern(ls, :normal)
+
+function convert_attribute(ls::Tuple{<:Union{Symbol,AbstractString},<:Any}, ::key"linestyle")
+    pattern(ls[1], ls[2])
+end
+
+function pattern(linestyle, gaps)
```

Comment:
can you call this something more obviously linestyle related? as names are visible across the whole codebase, I'd rather not have something so generic as `pattern`. Maybe `linestyle_pattern`, and for `diff_pattern` it could also be `linestyle_diff_pattern`

Reply:
what about putting `pattern`, `diff_pattern`, `check_pattern`, `convert_gaps` in a module instead?

Reply:
I'm for `line_pattern`, since they're 1D so it's mainly a line specific pattern.. The `style` seems a bit redundant in there.

Reply:
ok